doc/release-process.md

```diff
@@ -0,0 +1,14 @@
+# Release Process
+
+1. Open PR to master that
+   1. adds release notes to `doc/CHANGELOG.md` and
```

Comment:
In several projects, I've found it quite painful to write all the release notes at once when we actually do a release (the burden of writing the release notes falls on the guy who publishes the release and needs to document other people's additions to the codebase).

Something that worked much better for me was to incrementally update the future release notes: each PR to master that will impact the next release should update the future release notes. This way when we want to publish a release, we only need to do a bit of clean-up, but the release notes are already written and up-to-date. It needs a bit of effort at first to remind contributors to do it, but in the end we usually end up with better release notes.

Feel free to ignore if you think that doesn't make sense for this project, but I felt it was worth sharing in case it can be helpful.

Reply:
I think in principle I agree but I'm not really sure. My experience in this repo is that it happens quite often that you ask a  contributor to fix a nit before you ACK and then the PR is stalled for months. ;)

Reply:
Agree on @real-or-random's point but we can still remind regular contributors to do this. I wouldn't want to force this on irregular contributors because it's additional work that discourages from contributing PRs in the first place. We could add a pull request template that optionally requests a changelog update.

One significant downside of this is that pull requests would all edit similar space in the same file which, when one PR is merged, would require rebasing every other PR.

Reply:
> One significant downside of this is that pull requests would all edit similar space in the same file which, when one PR is merged, would require rebasing every other PR.

Apparently that's a real problem which led to a "crisis" in the GitLab project and needs an entirely non-trivial solution:
https://about.gitlab.com/blog/2018/07/03/solving-gitlabs-changelog-conflict-crisis/

I think then the maintainers should create the changelog... I suppose it will be easier for everyone (including maintainers).

Reply:
Bitcoin Core uses `doc/release-notes-<PRNUMBER>.md` files which are added by individual PRs, and are then merged into a release release-notes.md by maintainers.

Reply:
Just for reference, here's some docs on the process:
https://github.com/bitcoin/bitcoin/blob/23ae7931be50376fa6bda692c641a3d2538556ee/doc/release-process.md#write-the-release-notes
https://github.com/bitcoin/bitcoin/blob/23ae7931be50376fa6bda692c641a3d2538556ee/doc/release-notes.md

This has some helpful notes, e.g., a script to generate a (draft) changelog from git history. This may be helpful to us.